From 387d3b6fba95cb47c4dacc6bcd330148a9168850 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 7 Oct 2020 12:54:03 -0700
Subject: [PATCH] xlat: Mark IPPROTO_MAX last in IPPROTO_* constants

* xlat/inet_protocols.in (IPPROTO_MAX): It should be the last entry
  after adding IPPROTO_MPTCP this should have new value as
  IPPROTO_MPTCP + 1.

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Submitted [https://lists.strace.io/pipermail/strace-devel/2020-October/010253.html]
---
 xlat/inet_protocols.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/xlat/inet_protocols.in
+++ b/xlat/inet_protocols.in
@@ -32,5 +32,5 @@ IPPROTO_UDPLITE		136
 IPPROTO_MPLS		137
 IPPROTO_ETHERNET	143
 IPPROTO_RAW		255
-IPPROTO_MAX		256
 IPPROTO_MPTCP		262
+IPPROTO_MAX		263
--- a/xlat/inet_protocols.h
+++ b/xlat/inet_protocols.h
@@ -234,19 +234,19 @@ DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #else
 # define IPPROTO_RAW 255
 #endif
-#if defined(IPPROTO_MAX) || (defined(HAVE_DECL_IPPROTO_MAX) && HAVE_DECL_IPPROTO_MAX)
+#if defined(IPPROTO_MPTCP) || (defined(HAVE_DECL_IPPROTO_MPTCP) && HAVE_DECL_IPPROTO_MPTCP)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
-static_assert((IPPROTO_MAX) == (256), "IPPROTO_MAX != 256");
+static_assert((IPPROTO_MPTCP) == (262), "IPPROTO_MPTCP != 262");
 DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #else
-# define IPPROTO_MAX 256
+# define IPPROTO_MPTCP 262
 #endif
-#if defined(IPPROTO_MPTCP) || (defined(HAVE_DECL_IPPROTO_MPTCP) && HAVE_DECL_IPPROTO_MPTCP)
+#if defined(IPPROTO_MAX) || (defined(HAVE_DECL_IPPROTO_MAX) && HAVE_DECL_IPPROTO_MAX)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
-static_assert((IPPROTO_MPTCP) == (262), "IPPROTO_MPTCP != 262");
+static_assert((IPPROTO_MAX) == (263), "IPPROTO_MAX != 263");
 DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #else
-# define IPPROTO_MPTCP 262
+# define IPPROTO_MAX 263
 #endif
 
 #ifndef XLAT_MACROS_ONLY
@@ -353,12 +353,12 @@ static const struct xlat_data inet_proto
  XLAT(IPPROTO_RAW),
  #define XLAT_VAL_32 ((unsigned) (IPPROTO_RAW))
  #define XLAT_STR_32 STRINGIFY(IPPROTO_RAW)
- XLAT(IPPROTO_MAX),
- #define XLAT_VAL_33 ((unsigned) (IPPROTO_MAX))
- #define XLAT_STR_33 STRINGIFY(IPPROTO_MAX)
  XLAT(IPPROTO_MPTCP),
- #define XLAT_VAL_34 ((unsigned) (IPPROTO_MPTCP))
- #define XLAT_STR_34 STRINGIFY(IPPROTO_MPTCP)
+ #define XLAT_VAL_33 ((unsigned) (IPPROTO_MPTCP))
+ #define XLAT_STR_33 STRINGIFY(IPPROTO_MPTCP)
+ XLAT(IPPROTO_MAX),
+ #define XLAT_VAL_34 ((unsigned) (IPPROTO_MAX))
+ #define XLAT_STR_34 STRINGIFY(IPPROTO_MAX)
 };
 const struct xlat inet_protocols[1] = { {
  .data = inet_protocols_xdata,
